_posts/en/newsletters/2020-06-10-newsletter.md

```diff
@@ -0,0 +1,387 @@
+      necessarily the best ones. This implies the need for eviction."
+  a1link="https://bitcoincore.reviews/16756#l-41"
+
+  q2="Why does Bitcoin Core protect selected peers from eviction?"
+  a2="We want to keep connections that have proven to be reputable and increase
+      the difficulty for a malicious party to evict and occupy all the inbound
+      connections. Therefore, a small number of peers are protected for each of
+      several distinct characteristics which are difficult to forge. In order to
+      partition a node, the attacker must simultaneously be better at all of
+      them than honest peers."
+  a2link="https://github.com/bitcoin-core-review-club/bitcoin/blob/pr16756/src/net.cpp#L846"
+
+  q3="Describe the algorithm for inbound peer eviction in Bitcoin Core."
+  a3="Select all inbound peers not on the `NO BAN` list and not already
+      scheduled for disconnection. Of these, remove (protect) the best peers
+      according to costly, difficult-to-forge attributes, which each apply
+      separately: netgroup, lowest minimum ping time, recently sent transactions
+      and blocks, desirable service flags, and long-lasting connections. Of the
+      remaining peers, choose one to evict from the netgroup with the most
+      connections."
+  a3link="https://github.com/bitcoin-core-review-club/bitcoin/blob/pr16756/src/net.cpp#L851"
+%}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [Bitcoin Core 0.20.0][] is the newest major release from the original
+  Bitcoin project.  This release's most notable user-visible improvements
+  include default bech32 addresses for RPC users (the GUI began
+  defaulting to bech32 in 0.19), the ability to configure RPC
+  permissions for different users and applications (see [Newsletter
+  #77][news77 rpcwhitelist]), some basic support for generating PSBTs in
+  the GUI ([#74][news74 psbtgui] & [#82][news82 psbtgui]), and a
+  `sortedmulti` addition to the [output script descriptor][topic
+  descriptors] language that can make it easier to watch multisig
+  addresses generated from lexicographically sorted keys, such as those
+  used with ColdCard's native multisig support ([#68][news68
+  sortedmulti]).
+
+    Many less visible improvements have also been made to the code in
+    order to eliminate bugs, enhance security, and prepare for future
+    changes.  One forward-looking change that's attracted some attention
+    is the addition of an `asmap` configuration setting that allows
+    using a separately-downloaded database to improve Bitcoin Core's
+    ability to avoid connecting to too many peers all running on a
+    network owned by the same group, increasing its resistance to
+    [eclipse attacks][topic eclipse attacks] such as the [Erebus
+    attack][] (see [Newsletter #83][news83 asmap]).  However, one of the
```

Comment:
Likely depends on who one talks to, but I'm not sure the terminology and description is ideal:

- The asmap feature and discussion specifically targeted Erebus: see https://github.com/bitcoin/bitcoin/pull/16702 and https://github.com/bitcoin/bitcoin/issues/16599.

- The researchers may be biased but do describe Erebus as a separate, third partitioning attack (see https://erebus-attack.comp.nus.edu.sg, section "How is the Erebus attack different from previous partitioning attacks?"): *There are two previous partitioning attacks, i.e., the Bitcoin hijacking (by Apostolaski et al., IEEE SP'17) and the Eclipse attack (by Heilman et al., USENIX Sec'15)*, followed by a table that details the distinguishing characteristics.

Reply:
Perhaps the question is whether "eclipse attacks" has become a generic term for partitioning attacks, separate from the Eclipse attack (2015).

Reply:
@jonatack Eclipse Attack is a term that predates Bitcoin.  Here's a paper from 2006: https://homepage.divms.uiowa.edu/~ghosh/eclipse.pdf .  I don't claim to be an expert, but I think eclipse attack is the generic name for isolating a node from any honest peer so that you can feed it misinformation.  Heilman's paper describe several *methods* for eclipsing a Bitcoin node, and the Erebus attack describes another method.  In short, I think the Erebus attack is a type of eclipse attack (and I previously updated the text to say that).

Reply:
> @jonatack Eclipse Attack is a term that predates Bitcoin. Here's a paper from 2006

TIL - thanks! Will read.